### PR TITLE
use containerd shims instead of krustlet

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ K8S_DOWNLOADS_DIR="/opt/kubernetes/downloads"
 UBUNTU_RELEASE=$(lsb_release -r -s)
 TELEPORTD_PLUGIN_DOWNLOAD_DIR="/opt/teleportd/downloads"
 TELEPORTD_PLUGIN_BIN_DIR="/usr/local/bin"
-KRUSTLET_VERSION="v0.0.1"
+CONTAINERD_WASM_VERSION="cb07e66"
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:16]: todo: replace with official build

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 2]
```diff
@@ -88,16 +88,18 @@ downloadCNI() {
     retrycmd_get_tarball 120 5 "$CNI_DOWNLOADS_DIR/${CNI_TGZ_TMP}" ${CNI_PLUGINS_URL} || exit $ERR_CNI_DOWNLOAD_TIMEOUT
 }
 
-downloadKrustlet() {
-    local krustlet_url="https://acs-mirror.azureedge.net/krustlet-wagi/${KRUSTLET_VERSION}/linux/amd64/krustlet-wagi"
-    local krustlet_filepath="/usr/local/bin/krustlet-wagi"
+downloadContainerdWasmShims() {
+    local containerd_wasm_url="https://kubernetesreleases.blob.core.windows.net/containerd-wasm-shims/${CONTAINERD_WASM_VERSION}/linux/amd64"
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:92]: todo: use acs-mirror

**Comment by alexeldeib** [comment]
please ignore stale krustlet references and focus on logic, I know this needs post-merge cleanup

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 3]
```diff
@@ -88,16 +88,18 @@ downloadCNI() {
...
 
-downloadKrustlet() {
-    local krustlet_url="https://acs-mirror.azureedge.net/krustlet-wagi/${KRUSTLET_VERSION}/linux/amd64/krustlet-wagi"
-    local krustlet_filepath="/usr/local/bin/krustlet-wagi"
+downloadContainerdWasmShims() {
+    local containerd_wasm_url="https://acs-mirror.azureedge.net/containerd-wasm-shims/${CONTAINERD_WASM_VERSION}/linux/amd64"
+    local containerd_wasm_filepath="/usr/local/bin"
     if [[ $(isARM64) == 1 ]]; then
-        krustlet_url="https://kubernetesreleases.blob.core.windows.net/krustlet-wagi/arm64/linux/arm64/krustlet-wagi"
+        containerd_wasm_url="https://acs-mirror.azureedge.net/containerd-wasm-shims/${CONTAINERD_WASM_VERSION}/linux/arm64"
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:95]: arm64 binary will exist in like 30 min but this will fail PR build

**Review by ganeshkumar5699 — APPROVED** [review]

**Pull request merged**